test/wasm2js/func_ptrs.2asm.js

```diff
@@ -108,7 +107,14 @@ function asmFunc(global, env, buffer) {
   return FUNCTION_TABLE[i | 0]() | 0 | 0;
  }
  
- var FUNCTION_TABLE = [t1, t2, t3, u1, u2, t1, t3];
+ var FUNCTION_TABLE = [];
+ FUNCTION_TABLE[0] = t1;
+ FUNCTION_TABLE[1] = t2;
+ FUNCTION_TABLE[2] = t3;
+ FUNCTION_TABLE[3] = u1;
+ FUNCTION_TABLE[4] = u2;
+ FUNCTION_TABLE[5] = t1;
+ FUNCTION_TABLE[6] = t3;
```

Comment:
This new pattern increases code size compared to before, and also may be slower (as it just assigns without setting the length first - may confuse a VM).

Reply:
OK I think I can find a way to restore this behavior at least for internal tables, and perhaps even for exported ones.

Note that this pattern has always historically been used in emscripten since its the only option available when the table is imported.    So its not technically a code size regression for emscripten :+1: 

Do you thinks its worth pursuing this optimization?

Reply:
Oh, did we always do the unoptimal thing in emscripten? I forgot that...

Can leave for a followup, then.

But it should be easy to do, as JS lets you do convenient stuff like `FUNCTION_TABLE.push(foo, bar, baz, ..)`.

Reply:
I restored the old behaviour.

Also, this should increase the code size wins for https://github.com/emscripten-core/emscripten/pull/12291 even more!